### PR TITLE
#30 styled single list item

### PR DESCRIPTION
**Pull request by TurnbulA**
Gave the list a basic style in relation to design 
included buttons since they are part of the style

### CODE REVIEW

File: todos.js [hunk 1]
```diff
@@ -10,12 +10,17 @@ const renderJson = json => {
...
-          <li class ="json-section__completed">${userTask.completed}</li>
-        </ul>`
+        `<div class = "json-section__title">
+
+            <label class = "json-section__title--label">
+              <input type = "checkbox"/>
+              <span class = "check-custom"></span>
+              <div class ="json-section__title--text">
+              <p>${userTask.title}</p>
+            </div>
```
Comment (stephen-algeo) [inline todos.js]: text formatting

File: todos.js [hunk 2]
```diff
@@ -10,12 +10,17 @@ const renderJson = json => {
...
-          <li class ="json-section__userid">${userTask.userId}</li>
-          <li class ="json-section__id">${userTask.id}</li>
-          <li class ="json-section__title">${userTask.title}</li>
-          <li class ="json-section__completed">${userTask.completed}</li>
-        </ul>`
+        `<div class = "json-section__title">
+
+            <label class = "json-section__title--label">
+              <input type = "checkbox"/>
+              <span class = "check-custom"></span>
```
Comment (stephen-algeo) [inline todos.js]: don't use spaces

`class="check-custom"`

File: todos.js [hunk 3]
```diff
@@ -10,12 +10,17 @@ const renderJson = json => {
   const userTasks = json
     .map(
       userTask =>
```
Comment (stephen-algeo) [inline todos.js]: you can deconstruct the `userTask` here

File: style.css [hunk 4]
```diff
@@ -0,0 +1,92 @@
+html {
+  font-size: 10px;
```
Comment (mali891) [inline style.css:2]: Have a google of 'css resets', and see what might be useful to you

File: index.html [hunk 5]
```diff
@@ -4,11 +4,13 @@
...
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta http-equiv="X-UA-Compatible" content="ie=edge" />
-    <link rel="stylesheet" href="stye.css" />
+    <link rel="stylesheet" href="style.css" />
     <title>Document</title>
   </head>
   <body>
-    <div class="json-section"></div>
+    <div class="container">
+      <div class="json-section"></div>
```
Comment (mali891) [inline index.html]: `json-section` could be better named, maybe something like `to-do-list-container`, and then perhaps each item can be something like `to-do-list-item`

File: style.css [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  position: absolute;
+  content: "";
+  left: 1.2rem;
+  top: 1.2rem;
+  height: 0;
+  width: 0;
+  border-radius: 0.5rem;
+  border: solid #8b8b8b;
+  border-width: 0 0.1rem 0.1rem 0;
+  -webkit-transform: rotate(0deg) scale(0);
```
Comment (mali891) [inline style.css]: What do these do?

File: style.css [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  background-color: #fd3d6c;
+  border-radius: 0.5rem;
+  -webkit-transform: rotate(0deg) scale(1);
+  -ms-transform: rotate(0deg) scale(1);
+  transform: rotate(0deg) scale(1);
+  opacity: 1;
+  border: 2px solid transparent;
+}
+
+.json-section__title--label input:checked ~ .check-custom::after {
```
Comment (mali891) [inline style.css]: What does `~` mean?

File: todos.js (hunk 2 above)
Comment (mali891) [inline todos.js]: Use BEM

File: style.css (hunk 6 above)
Comment (TurnbulA) [reply style.css]: Make the tick within the checkbox

File: style.css (hunk 7 above)
Comment (TurnbulA) [reply style.css]: The tilde (~) separates the two selectors

File: style.css [hunk 8]
```diff
@@ -15,32 +15,32 @@ body {
...
   margin-left: 5rem;
   text-transform: capitalize;
 }
-.todo-list--label {
+.todo-list-item__label {
   display: flex;
   align-items: center;
 }
-.todo-list--label input {
+.todo-list-item__label input {
```
Comment (mali891) [inline style.css]: `todo-list-item__input`

File: style.css [hunk 9]
```diff
@@ -15,32 +15,32 @@ body {
...
   position: absolute;
   opacity: 0;
   cursor: pointer;
 }
 
-.todo-list--label .todo--check-custom {
+.todo-list-item__label .todo-list-item__check-custom {
```
Comment (mali891) [inline style.css]: Just `.todo-list-item__check-custom` should suffice

File: style.css [hunk 10]
```diff
@@ -50,7 +50,7 @@ body {
   cursor: pointer;
 }
 
-.todo-list--label .todo--check-custom::after {
+.todo-list-item__label .todo--check-custom::after {
```
Comment (mali891) [inline style.css]: Just `.todo--check-custom::after` should suffice

File: style.css [hunk 11]
```diff
@@ -66,7 +66,7 @@ body {
   opacity: 1;
 }
 
-.todo-list--label input:checked ~ .todo--check-custom {
+.todo-list-item__label input:checked ~ .todo--check-custom {
```
Comment (mali891) [inline style.css]: `todo-list-item__input`

File: style.css [hunk 12]
```diff
@@ -76,7 +76,7 @@ body {
   border: 2px solid transparent;
 }
 
-.todo-list--label input:checked ~ .todo--check-custom::after {
+.todo-list-item__label input:checked ~ .todo--check-custom::after {
```
Comment (mali891) [inline style.css]: `todo-list-item__input`

File: todos.js (hunk 3 above)
Comment (mali891) [reply todos.js]: I second this

File: todos.js [hunk 13]
```diff
@@ -10,12 +10,17 @@ const renderJson = json => {
...
-        `<ul class="list">
-          <li class ="json-section__userid">${userTask.userId}</li>
-          <li class ="json-section__id">${userTask.id}</li>
-          <li class ="json-section__title">${userTask.title}</li>
-          <li class ="json-section__completed">${userTask.completed}</li>
-        </ul>`
+        `<div class = "todo-list-item">
+
+            <label class = "todo-list-item__input">
```
Comment (mali891) [inline todos.js]: Please remove the spaces here

File: todos.js [hunk 14]
```diff
@@ -10,12 +10,17 @@ const renderJson = json => {
   const userTasks = json
     .map(
       userTask =>
-        `<ul class="list">
-          <li class ="json-section__userid">${userTask.userId}</li>
-          <li class ="json-section__id">${userTask.id}</li>
-          <li class ="json-section__title">${userTask.title}</li>
-          <li class ="json-section__completed">${userTask.completed}</li>
-        </ul>`
+        `<div class = "todo-list-item">
```
Comment (mali891) [inline todos.js]: Spaces

File: todos.js (hunk 3 above)
Comment (mali891) [reply todos.js]: Feel free to ask about destructuring if you're not sure

Comment (stephen-algeo) [reply todos.js]: https://wesbos.com/destructuring-objects/

File: style.css [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+  transform: rotate(45deg) scale(1);
+  opacity: 1;
+  left: 0.8rem;
+  top: 0.3rem;
+  width: 0.6rem;
+  height: 1.2rem;
+  border: solid #ffffff;
+  border-width: 00 0.1rem 0.1rem 0;
+  background-color: transparent;
+  border-radius: 0;
```
Comment (mali891) [inline style.css]: Unnecessary

Comment (TurnbulA) [reply style.css]: Is necessary. Border radius is set to a different value elsewhere this converts the tick back to standard border radius

File: style.css [hunk 16]
```diff
@@ -1,8 +1,11 @@
 html {
   font-size: 10px;
+  font-family: "Rubik", sans-serif;
+  padding: 1rem;
```
Comment (mali891) [inline style.css]: Don't need this

File: style.css [hunk 17]
```diff
@@ -1,8 +1,11 @@
...
 }
 body {
   font-size: 1.6rem;
+  margin: 0;
```
Comment (mali891) [inline style.css]: Don't need this

**Review by stephen-algeo — DISMISSED** [review]

File: index.html [hunk 18]
```diff
@@ -13,6 +13,9 @@
   </head>
   <body>
     <div class="container">
+      <section class="todo-list-header">
```
Comment (mali891) [inline index.html:16]: The header shouldn't be part of a single list item PR

File: style.css [hunk 19]
```diff
@@ -1,16 +1,18 @@
...
 }
 body {
   font-size: 1.6rem;
-  margin: 0;
 }
 .container {
   max-width: 120rem;
   margin: 0 auto;
 }
+.todo-list-header {
```
Comment (mali891) [inline style.css:12]: Header shouldn't be part of a PR for single list item

File: index.html (hunk 18 above)
Comment (mali891) [reply index.html:16]: Also, use BEM

File: resources/Check.svg [hunk 20]
```diff
@@ -0,0 +1,10 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="27.788" height="19.548" viewBox="0 0 27.788 19.548">
+  <defs>
+    <style>
+      .cls-1 {
```
Comment (mali891) [inline resources/Check.svg:4]: SVG definitions can be set in the CSS file rather than inline

File: resources/Close.svg [hunk 21]
```diff
@@ -0,0 +1,10 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30">
+  <defs>
+    <style>
+      .cls-1 {
```
Comment (mali891) [inline resources/Close.svg:4]: SVG definitions can be set in the CSS file rather than inline

**Review by mali891 — APPROVED** [review]

**Pull request merged**